src/routing/comm.rs

```diff
@@ -424,15 +432,10 @@ mod tests {
 
         // Receive one message and drop the incoming stream.
         {
-            if let Some(mut incoming_msgs) =
-                time::timeout(TIMEOUT, recv_incoming_connections.next()).await?
-            {
-                if let Some(msg) = time::timeout(TIMEOUT, incoming_msgs.next()).await? {
-                    assert_eq!(msg.get_message_data(), msg0);
-                    msg0_received = true;
-                }
+            if let Some((_src, msg)) = time::timeout(TIMEOUT, incoming_msgs.next()).await? {
+                assert_eq!(msg, msg0);
+                msg0_received = true;
```

Comment:
To keep the original purpose of this test I think we need to now explicitly disconnect from the peer here: 

```rust
recv_enpoint.disconnect_from(src)
```
Also change the comment above to something like: "Receive one message and disconnect from the peer"